R/prep_access.R

```diff
@@ -0,0 +1,40 @@
+#' Prepare access
+#'
+#' Extract variableNames for a given data file and add them to the attributes.csv
+#' @param data_path path to the data folder. Defaults to "data" and R 'data' file types
+#' @param access_path path to the access.csv file. Defaults to "data/metadata/access.csv".
+#'
+#' @return the functions writes out the updated access.csv file to access_path.
+#' @export
+prep_access <- function(data_path = here::here("data"),
+                        access_path = here::here("data", "metadata",
+                                                 "access.csv")
+                        ){
+
+  if(!file.exists(data_path)){stop("invalid path to data folder")}
+  if(!file.exists(access_path)){
+    stop("access file does not exist. Check path or run create_spice?")}
+
+  access <- readr::read_csv(access_path, col_types = readr::cols())
+
+  # read file info
+  fileNames <- tools::list_files_with_type(data_path, "data", full.names = TRUE)
```

Comment:
Nice find on this function! I see that it's a bit strict as to what it considers data but this is fine for now since we're only supporting CSV:

https://github.com/SurajGupta/r-source/blob/a28e609e72ed7c47f6ddfbb86c85279a0750f0b7/src/library/tools/R/utils.R#L1409-L1416

Reply:
I was wondering what "data" type meant, good find. Am I reading that correctly that .tsv is missing? currently `prep_attributes` includes both csv and tsv

Reply:
Yeah sounds right. File extensions are a very rough way to detect file format so in the end, the user should be able to say "hey! my file _is_ tabular!" regardless of what automated routine we set up. providing a helpful warning like, "Hey, it looked like you tried to prep from files we don't support. If they truly are CSV or TSV, re-run the previous command with `ignore_extension = TRUE` to attempt auto-detecting your headers".